Fix Azure SharedConfig xml parsing in Attribute retrieval

jepio opened the pull request:
serde-xml-rs 0.8.0 stopped parsing XML attributes into fields, unless the name starts with '@', so afterburn now panics on Azure.
Add a mock/test that exercises that part of the parser and fix the bug by renaming the 3 fields that seem to be affected by this.

See #1231

Commit:
microsoft/azure: Mock goalstate.SharedConfig output in tests

There are currently no tests exercising the code inside
get_attributes(). Extend the mock_tests to return a shared config
document at the path specified in the goalstate document so that this
code can be tested.

Signed-off-by: Jeremi Piotrowski <jpiotrowski@microsoft.com>

Commit:
microsoft/azure: Fix SharedConfig parsing of XML attributes

serde-xml-rs 0.8.0 introduced a breaking change by requiring that fields
deserialized from attributes must have a name starting with '@'. Rename
the fields that are effected by this. This is now testable thanks to the
just added SharedConfig mock.

Signed-off-by: Jeremi Piotrowski <jpiotrowski@microsoft.com>

Commit:
docs/release-notes: Add entry for Azure SharedConfig XML parsing fix

Signed-off-by: Jeremi Piotrowski <jpiotrowski@microsoft.com>

jlebon commented on src/providers/microsoft/azure/mock_tests.rs:
```diff
@@ -234,6 +266,8 @@ fn test_vmsize() {
     let r = attributes.get("AZURE_VMSIZE");
```
Let's add a check here for an attribute which comes from the shared config, e.g. `AZURE_IPV4_VIRTUAL`?

jepio replied:
Done

jlebon reviewed (approved):
Thanks! One comment but this looks sane to me.

Commit:
microsoft/azure: Add XML attribute alias for serde-xml-rs Fedora compat

The Fedora build uses a different serde-xml-rs version than the one in
the Cargo.lock file. That version doesn't support the '@identifier'
notation for attributes. Specify an alias to make deserialization work
with either serde-xml-rs version.

Signed-off-by: Jeremi Piotrowski <jpiotrowski@microsoft.com>
